proposals/buildfile.md

```diff
@@ -0,0 +1,187 @@
+# Jib CLI Buildfile Specification
+
+Specification for a YAML buildfile describing building a container image. This buildfile can be
+used by the jib-cli to generate a container. It is translated directly into a buildplan and
+passed to the builder.
+
+```yaml
+# "FROM" with option to use os/architecture for manifest lists
+from: "gcr.io/distroless/java:8"
+
+# "FROM" with detail for manifest lists or multiple architectures
+from:
+  image: "gcr.io/distroless/java:8"
+  platforms:
+    - architecture: "arm"
+      os: "linux"
+      os.version: "a"
+      os.features:
+        - "b1"
+        - "b2"
+      variant: "c"
+      features:
+        - "d1"
+        - "d2"
+    - architecutre: amd64
+      os: darwin
+
+creationTime: 0 # millis since epoch or iso8601 creation time
+format: Docker # Docker or OCI
+
+environment:
+  "MY_KEY1": "value"
+  "MY_KEY2": "value"
+labels:
+  "com.example.owner": "person"
+  "com.example.mode": "dev"
+volumes:
+  - "/myvolume"
+  - "/youvolume"
+
+exposedPorts:
+  - "123/udp"
+  - "456"
+  - "789/tcp"
+
+user: root
+workingDirectory: "/somewhere"
+entrypoint:
+  - "java"
+  - "-jar"
+cmd:
+  - "myjar.jar"
+
+# global file properties applied to all layers
+layerProperties:
+  filePermissions: "644"
+  directoryPermissions: "755"
+  user: "0"
+  group: "0"
+  timestamp: "0"
+layers:
+  - name: "scripts and classes"
+    # file properties only applied to this layer "scripts and classes"
+    properties:
+      filePermissions: "333"
+      directoryPermissions: "777"
+      user: "goose"
+      group: "3"
+      timestamp: "2020-06-03T19:31:50+00:00"
+    files:
+      - from: "target/scripts"
```

Comment:
file copy behavior needs to be defined

Reply:
target/bin/script.amd64
->
/app/bin/script (script is a file)

Reply:
/app/bin/script/ is a directory
/app/bin/script is a file, if from is a file
/app/bin/script is a directory, if from is a directory

Reply:
what about the behavior of docker add/copy?

Reply:
symlinks?
- expand out symlinks, they will be regular files on destination.
- preserve if contained within the layer

alternatives
- preserve and hope the file exists on the destination system?

symlink features: expand, preserve (check rsync behavior)